### PR TITLE
[DOC] Update Sphinx & misc doc related things

### PR DESCRIPTION
**Pull request by TomDonoghue**
This pull request is partially related to #167, and also includes various other updates.

Updates included:
- Tutorial tweaks & updates, including fixing tutorial 1 (comments in #167)
   - Notably this updates and makes more consistent the math. For consistency, power is considered in log-spacing (with `NPS` and `L` now consistently reflecting log-spaced power values) 
- Update format of the data objects
- Sphinx updates:
    - update to use to use autosummary / autodoc templates. This renders some API documentation better (including for data objects), and relatedly cleans things up and fixes linking between API docs & galleries, and also adds the "link from API to galleries they are used it"
- Doc updates for website, including updating the references page, and updating the API page
- Converts the README to rst format. This drops our requirement of m2r for building the docsite (which is good because m2r is no longer supported)
- Add the sphinx copybutton extension to the website
- Set up a `doctest` procedure, and add a bunch of docstring examples that follow doctest.
- Updates for `fooof.analysis.periodic` that organizes docstring and tutorials to guide usage
- Code fixes: bug fixes in `fooof.sim.gen`, a couple moves of internal functitons (for organizational purposes), and small updates to managing plot settings and file name extension handling
- Update and add example data, including adding a new demo data file of a group of power spectra, and related update to use this in the tutorial (so that the tutorial uses more real data)
- Add a new file to generate all the reports & images used in the documentation site, using the example data from the repository. Update the images in the repository.

### CODE REVIEW

**Comment by TomDonoghue** [comment]
Okay, team - review time! This is (another) sweep of the documentation. 

For ease of review, the updated documentation site reflect all these changes is now hosted, so you can check the the rendered version on the docsite (https://fooof-tools.github.io/fooof/) to see what these changes actually look. 

- @ryanhammonds : I've tagged you to do a full review this PR! We can perhaps chat more sphinx things and/or review protocols before you jump into it though!
- @nschawor : this update includes the fixes / updates to documenting the periodic analysis functions
- @rdgao : tutorial 1 math updated. It *should* be consistent (I chose to keep NPS & L as log). It would be great if you could throw a quick eye on it and check this addresses your comments without messing anything up :)
- @voytek : nothing in particular needed, just tagging in case you want to have a look

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+    # Initialize a FOOOF model object with defined settings
+    fm = FOOOF(peak_width_limits=[1.0, 8.0], max_n_peaks=6, min_peak_height=0.1,
+               peak_threshold=2.0, aperiodic_mode='fixed')
+
+**Fitting a Group of Power Spectra**
+
+Next is an example workflow for fitting a group of neural power spectra.
+In this case, 'freqs' is again a 1D array of frequency values, and 'spectra' as a 2D array of power spectra.
+We can fit this  which can be fit as:
```
Comment (ryanhammonds) [inline README.rst:249]: Small typo here

**Comment by TomDonoghue** [comment]
Small note to self for when updating after review:
- the reference to `backrefs` can be removed from the `clean` command in the Makefile

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+.. _ProjectStatus: https://www.repostatus.org/#active
+
+.. |Version| image:: https://img.shields.io/pypi/v/fooof.svg
+.. _Version: https://pypi.python.org/pypi/fooof/
+
+.. |BuildStatus| image:: https://travis-ci.org/fooof-tools/fooof.svg
+.. _BuildStatus: https://travis-ci.org/fooof-tools/fooof
+
+.. |codecov| image:: https://codecov.io/gh/fooof-tools/fooof/branch/master/graph/badge.svg
+.. _codecov : https://codecov.io/gh/fooof-tools/fooof)
```
[the remark this answers is not in the document]
Comment (TomDonoghue) [reply README.rst:17]: Oops, yeh - good catch!

File: fooof/analysis/periodic.py [hunk 3]
```diff
@@ -73,7 +110,7 @@ def get_band_peak_group(peak_params, band, n_fits, threshold=None, thresh_param=
         Peak parameters, for a group fit, with shape of [n_peaks, 4].
     band : tuple of (float, float)
         Frequency range for the band of interest.
-        Defind as: (lower_frequency_bound, upper_frequency_bound).e
+        Defined as: (lower_frequency_bound, upper_frequency_bound).e
```
[the remark this answers is not in the document]
Comment (TomDonoghue) [reply fooof/analysis/periodic.py:113]: yep.

File: fooof/bands/bands.py [hunk 4]
```diff
@@ -12,6 +12,12 @@ class Bands():
     ----------
     bands : dict
         Band definitions. Each entry should be {'label' : (f_low, f_high)}.
+
+    Examples
+    --------
+    Define a bands object object storing canonical frequency bands:
```
[the remark this answers is not in the document]
Comment (TomDonoghue) [reply fooof/bands/bands.py:18]: Well, you know, it's just twice as object-y.

File: fooof/tests/analysis/test_periodic.py [hunk 5]
```diff
@@ -38,13 +38,16 @@ def test_get_band_peak():
     assert np.all(np.isnan(get_band_peak(data, [4, 8])))
 
     # Test multiple results - return all
-    assert np.array_equal(get_band_peak(data, [10, 15], ret_one=False), [[10, 1, 1.8], [14, 2, 4]])
+    assert np.array_equal(get_band_peak(data, [10, 15], select_highest=False),
+                          np.array([[10, 1, 1.8], [14, 2, 4]]))
```
[the remark this answers is not in the document]
Comment (TomDonoghue) [reply fooof/tests/analysis/test_periodic.py:42]: definitely could be - but I don't think we need to change this here, as is matches the explicit output of the other tests.

**Pull request merged**